packages/rolldown/src/options/bindingify-output-options.ts

```diff
@@ -75,20 +75,13 @@ function bindingifySourcemap(
 function bindingifyEsModule(
   esModule: NormalizedOutputOptions['esModule'],
 ): BindingOutputOptions['esModule'] {
-  switch (esModule) {
-    case 'always':
-    case true:
-      return 'always'
-
-    case 'never':
-    case false:
-      return 'never'
-
-    case 'if-default-prop':
-    case undefined:
-      return 'if-default-prop'
+  if (typeof esModule === 'boolean') {
+    return esModule
+  }
```

Comment:
I think you just need to remove:

```js
    case 'never':
```

and:

```js
    case 'always':
```

Reply:
In the configuration file, we can simply remove the boolean value. During the normalization step on the JavaScript side, we only need to transform the boolean to the `’always’ | ‘never’`. On the Rust side, let’s maintain the current value to prevent panic.

Reply:
Yes, I understand what you mean. 
I thought the modification on the `binding crate` side was feasible, as mentioned by  @hyf0, but it seems to have encountered the problem you encountered. I don't know if it's a problem with my code writing.

Reply:
Actually, in the first prototype of the previous PR, I also tried it. Sadly, I encountered a similar panic:).